core/trietxn_test.go

```diff
@@ -0,0 +1,85 @@
+package core
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/NethermindEth/juno/db"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/bits-and-blooms/bitset"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTrieTxn(t *testing.T) {
+	db := db.NewTestDb()
+	prefix := []byte{37, 44}
+
+	key := bitset.New(44)
+	node := new(TrieNode)
+
+	// put a node
+	if err := db.Update(func(txn *badger.Txn) error {
+		tTxn := &TrieTxn{txn, prefix}
+
+		value, _ := new(felt.Felt).SetRandom()
+		node.UnmarshalBinary(value.Marshal())
+
+		return tTxn.Put(key, node)
+	}); err != nil {
+		t.Error(err)
+	}
```

Comment:
These should be separated into separate subsets so that if one of them fails we can see immediately which one failed. So please wrap all the cases in `t.Run(caseName, func(t *t.testing.T)`